src/components/Combobox/Common/useCombobox.tsx

```diff
@@ -40,28 +40,52 @@ export const useCombobox = <T extends Option, V extends string | Option>({
   onBlur?: (e: FocusEvent<HTMLInputElement, Element>) => void;
 }) => {
   const [inputValue, setInputValue] = useState<string>("");
+  const [highlightedIndex, setHighlightedIndex] = useState<number | undefined>(
+    -1
+  );
   const [active, setActive] = useState(false);
   const typed = Boolean(selectedItem || active || inputValue);
 
   const itemsToSelect = getItemsFilter<T>(inputValue, options);
 
+  useEffect(() => {
+    // Skip when no selected item
+    if (!selectedItem || !highlightedIndex || highlightedIndex > -1) {
+      return;
+    }
+
+    // Find hilighted index in items to select base on selected item value
+    // or label. Label is used as fallback when API return  item value as Id
+    const index = itemsToSelect.findIndex(
+      (item) =>
+        item.value === selectedItem?.value ||
+        item.value === selectedItem?.label?.toLocaleLowerCase()
+    );
+
+    setHighlightedIndex(index);
+  }, [highlightedIndex, itemsToSelect, selectedItem]);
+
   const {
     isOpen,
     getToggleButtonProps,
     getLabelProps,
     getMenuProps,
     getInputProps: _getInputProps,
-    highlightedIndex,
     getItemProps,
   } = useDownshiftCombobox({
     items: itemsToSelect,
     itemToString: (item) => item?.label ?? "",
     selectedItem,
+    highlightedIndex,
+    onHighlightedIndexChange: ({ highlightedIndex }) => {
+      setHighlightedIndex(highlightedIndex);
```

Comment:
issue: isn't cause the loop in use effect? seems like it will also trigger useEffect and re-set index once again

Reply:
+ you never use `highlightedIndex` in the effect